### ISSUE TITLE
Latest Eval file Loading Issue. Engine Crash

### ISSUE THREAD

**Issue by ExtremeMoves** [amchess/ShashChess#52]
I got a problem with the ShashChess 34.6 as i am not able to load in newer eval files.  The newest eval file that will load up without crashing the engine is nn-5af11540bbfe.nnue which was released quite a few months ago. I tested almost all eval files that is newer than the one i listed above and all of them failed to load crashing the engine.  The following below was captured as an "informational" item listed in eventviewer.  Not sure if the below error log will help you or not but its posted below.

Fault bucket 1782316213243264143, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: ShashChess34.6-x86-64-avx2.exe
P2: 0.0.0.0
P3: 10.0.19045.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\dude\AppData\Local\Temp\RDR4A2C.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER4A4D.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER4A5D.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER4A5F.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER4A8F.tmp.txt

These files may be available here:

Analysis symbol: 
Rechecking for solution: 0
Report Id: a261876e-87a3-432c-ac5e-e8ea8927ef2f
Report Status: 268435456
Hashed bucket: d9049aec94e5f45478bc0f271fc8bc8f
Cab Guid: 0

**Comment by amchess** [comment]
If the architecture is different, you can't.

**Comment by ExtremeMoves** [comment]
Well i understand that but the problem here is that Stockfish has moved on to better network architectures and have been making newer networks as far back as last summer in 2023.  I thought ShashChess would have stayed in step with Stockfish builds that use the newer network files. All the newer stockfish builds can read any of the last networks going back many months ago and recent ones as well but ShashChess cannot read these newer files apparently.  So i am not sure what else i can say about it. Unless code is added that can allow ShashChess to read in newer network versions?

**Comment by ExtremeMoves** [comment]
I think i know what you will say as i now remembered a post you made a while back in regards to stockfish updates.  If i understand this correctly you will not update the core of the engine until the next major release of stockfish.  Because of this unfortunately i see that right after the major release of Stockfish 16 they did a small update on July 1, 2023 and added the following.

"[Update NNUE architecture to SFNNv7 with larger L1 size of 2048](https://github.com/official-stockfish/Stockfish/commit/915532181f11812c80ef0b57bc018de4ea2155ec)"

 So basically any network make beyond June 30, 2023 will not load into ShashChess untill they release Stockfish 17 and then ShashChess will get a major update along with any other modifications from the ShashChess team.

**Comment by amchess** [comment]
Exactly
